whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotRecordConsumer.java

```diff
@@ -0,0 +1,78 @@
+package net.ripe.db.nrtm4;
+
+import com.google.common.collect.Maps;
+import net.ripe.db.nrtm4.domain.NrtmDocumentType;
+import net.ripe.db.nrtm4.domain.NrtmVersionInfo;
+import net.ripe.db.nrtm4.domain.NrtmVersionRecord;
+import net.ripe.db.nrtm4.domain.SnapshotFileRecord;
+import net.ripe.db.whois.common.domain.CIString;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import static net.ripe.db.nrtm4.SnapshotRecordProducer.POISON_PILL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class SnapshotRecordConsumer implements Supplier<Map<CIString, byte[]>> {
```

Comment:
Is the class a Consumer or a Supplier? The class name is confusing

Reply:
Renamed classes